app.ts

```diff
@@ -0,0 +1,193 @@
+//Loads the table to the screen
+window.onload = () => GetTotalNumberOfRecords();
+//Resizes page during window resizing
+window.onresize = () => Resize();
+
+//Variable declarations
+let totalRecordCount: number;
+let columnNames: any[];
+let records: any[];
+let from: number = 0;
+let to: number = 0;
+let NumberOfRows = Math.floor((window.innerHeight - 50) / 24) - 1;
+let timeout: any;
+
+//Function to retrieve total amount of records
+function GetTotalNumberOfRecords() {
+  $.ajax({
+    url: "/recordCount",
+    type: "GET",
+    timeout: 12000,
+    dataType: "text"
+  })
+    .done(function(responseText: any) {
+      totalRecordCount = responseText;
+      console.log(totalRecordCount);
+      Resize();
+    })
+
+    .fail(function() {
+      alert("Could not retrieve total record number");
+    });
+}
+
+//function to retrieve column names
+function GetColumnNames() {
+  $.ajax({
+    url: "/columns",
+    type: "GET",
+    timeout: 12000,
+    dataType: "json"
+  })
+    .done((responsejson: any) => {
+      columnNames = responsejson;
+      GetActualRecordsAmount();
+    })
+    .fail(() => {
+      alert("Could not display column names");
+    });
+}
+
+//Function to retrieve the record data
+function GetActualRecordsAmount() {
+  $.ajax({
+    url: "/records?from=" + from + "&to=" + to,
+    timeout: 12000,
+    dataType: "JSON"
+  })
+    .done((responseRecords: any) => {
+      records = responseRecords;
+      BuildTable();
+    })
+    .fail(() => {
+      alert("No available records for your selection");
+    });
+}
+
+//Function builds the table and populates with data from the API calls
+function BuildTable() {
+  if (document.getElementById("Table") != null) {
+    $("Table").remove();
+  }
+
+  const body = document.getElementsByTagName("body")[0];
+
+  const ourTable = document.createElement("table");
+  ourTable.setAttribute("id", "Table");
+  //ourTable.style.borderCollapse = "collapse";
+  ourTable.style.borderSpacing = "0";
+
+  const thead = document.createElement("thead");
+  ourTable.appendChild(thead);
+
+  //Column names are created for the table
+  for (let c = 0; c < columnNames.length; c++) {
+    thead
+      .appendChild(document.createElement("th"))
+      .appendChild(document.createTextNode(columnNames[c]));
+  }
+
+  //Cells are created and populated with data
+  for (let i = 0; i < records.length; i++) {
```

Comment:
Use new for loop methods from style guide

i.e. `for (const key in records) { ... }`
or `for (const record of records) { ... }`